sdk/containerregistry/azure-containerregistry/azure/containerregistry/_anonymous_exchange_client.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import re
+from typing import TYPE_CHECKING, Dict, Any
+
+from ._exchange_client import ExchangeClientAuthenticationPolicy
+from ._generated import ContainerRegistry
+from ._helpers import _parse_challenge
+from ._user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+
+PASSWORD = u"password"
+
+
+class AnonymousACRExchangeClient(object):
+    """Class for handling oauth authentication requests
+
+    :param endpoint: Azure Container Registry endpoint
+    :type endpoint: str
+    :param credential: Credential which provides tokens to authenticate requests
+    :type credential: :class:`~azure.core.credentials.TokenCredential`
+    """
+
+    BEARER = "Bearer"
+    AUTHENTICATION_CHALLENGE_PARAMS_PATTERN = re.compile('(?:(\\w+)="([^""]*)")+')
```

Comment:
These seem to be defined on all 4 ExchangeClients - perhaps they could be refactored out somewhere?
Not a big deal - doesn't need to be changed now.

Reply:
They can all be removed, I parse the challenge policy in a separate file with everything properly scoped